client/common/configuration/templates/TranslationDirective.ts

```diff
@@ -0,0 +1,57 @@
+import { PartInfo, PartType, directive } from 'lit/directive.js'
+import { AsyncDirective } from 'lit/async-directive.js'
+import { RegisterClientHelpers } from '@peertube/peertube-types/client';
+import { unsafeHTML } from 'lit/directives/unsafe-html.js';
+import { html } from 'lit';
+import { unsafeStatic } from 'lit/static-html.js';
+
+export class TranslationDirective extends AsyncDirective {
+
+    private _peertubeHelpers?: RegisterClientHelpers
+
+    private _translatedValue : string = ''
+    private _localizationId : string = ''
+
+    private _allowUnsafeHTML = false
+
+    constructor(partInfo: PartInfo) {
+        super(partInfo);
+
+        //_peertubeOptionsPromise.then((options) => this._peertubeHelpers = options.peertubeHelpers)
+    }
+
+    // update = (part: ElementPart) => {
+    //     if (part) console.log(`Element : ${part?.element?.getAttributeNames?.().join(' ')}`);
+    //     return this.render(this._localizationId);
+    // }
+
+    override render = (locId: string, allowHTML: boolean = false) => {
+        this._localizationId = locId // TODO Check current component for context (to infer the prefix)
+
+        this._allowUnsafeHTML = allowHTML
+
+        if (this._translatedValue === '') {
+            this._translatedValue = locId
+        }
+
+        this._asyncUpdateTranslation()
```

Comment:
I think this will not pass linting: async calls must be handled (at least, add `.then(() => {}, () => {})`)

Reply:
Oh ok I'll add an error resolution. What about translate calls?

Reply:
There is an await, so it is ok. Check https://typescript-eslint.io/rules/no-misused-promises/ for more information